File: 94_Binary Tree Inorder Traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
    result := [] int{}
    stack := [] *TreeNode{}
    
    curr := root
    
    for curr != nil || len(stack) != 0 {
        for curr != nil {
            stack = append(stack, curr)
            curr = curr.Left
        }
        
        curr = stack[len(stack) - 1]
        stack = stack[:len(stack) - 1]
        result = append(result, curr.Val)
        curr = curr.Right
    }
    return result
}

File: 94_Binary Tree Inorder Traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    if(!root) {
        return [];
    }
    
    const result = [];
    let stack = [];
    
    let curr = root;
    
    while(curr || stack.length) {
        while(curr) {
            stack.push(curr);
            curr = curr.left;
        }
        
        curr = stack.pop();
        result.push(curr.val);
        curr = curr.right;
    }
    return result;
};